cds-snc/notification-api
# Fix fake callback

# Summary | Résumé

The `INTERNAL_TEST_NUMBER` (which is used by the smoke test) was causing an error after we switched on Pinpoint: the fake callback was being processed by the sns callback task rather than the pinpoint callback task. This PR ensures that the correct task (ie based on the provider being used) is called.

## Related Issues | Cartes liées

* https://app.zenhub.com/workspaces/notify-planning-core-6411dfb7c95fb80014e0cab0/issues/gh/cds-snc/notification-planning-core/360

# Test instructions | Instructions pour tester la modification

To test locally:

Set env vars (from staging):
```
AWS_PINPOINT_SC_POOL_ID = "pool-b20333ce1e4e49309ba1db3bf94a3f57"
AWS_PINPOINT_DEFAULT_POOL_ID = "pool-8885654b47d6466f9aaee14c62494ff2"
```

Send an SMS to 16135550123. It should immediately register as sent. and there should be no celery errors.

# Release Instructions | Instructions pour le déploiement

None.

# Reviewer checklist | Liste de vérification du réviseur


> ⚠ If boxes cannot be checked off before merging the PR, they should be moved to the "Release Instructions" section with appropriate steps required to verify before release. For example, changes to celery code may require tests on staging to verify that performance has not been affected.
